python4j/python4j-numpy/src/test/java/PythonNumpyMultiThreadTest.java

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Konduit K.K.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+import org.eclipse.python4j.*;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+
+@NotThreadSafe
+@RunWith(Parameterized.class)
+public class PythonNumpyMultiThreadTest {
+    private DataType dataType;
+
+    public PythonNumpyMultiThreadTest(DataType dataType) {
+        this.dataType = dataType;
+    }
+
+    @Parameterized.Parameters(name = "{index}: Testing with DataType={0}")
+    public static DataType[] params() {
+        return new DataType[]{
+//                DataType.BOOL,
+//                DataType.FLOAT16,
+//                DataType.BFLOAT16,
+                DataType.FLOAT,
+                DataType.DOUBLE,
+//                DataType.INT8,
+//                DataType.INT16,
+                DataType.INT32,
+                DataType.INT64,
+//                DataType.UINT8,
+//                DataType.UINT16,
+//                DataType.UINT32,
+//                DataType.UINT64
```

Comment:
Are these test cases disabled for a reason, or just forgotten from debugging?

Reply:
Forgotten from debugging. Will enable in next pr.